File: BEGINNER/2028 Sequência de Sequência.py
# algoritmo
# VARIOS CASOS DE TESTE
# PROGRAMA TERMINA COM EOF
# RECEBER O VALOR N DA SEQUENCIA
# O NUMERO 0 EQUIVALE A 1 NUMERO
# A SOMA DE TODOS OS NUMEROS RESULTA NO TOTAL DE NUMEROS DA SEQUENCIA (0 + 1 = 2 NUMEROS)
# PASSO - PASSO E REGRAS
# OBS: ADICIONAR OS NUMEROS DA SEQUENCIA EM UMA LISTA
# PASSO 1: INICIAR UM LOOP QUE VAI RECEBER N NUMEROS DA SEQUENCIA PARA FAZER UMA CONTAGEM
# PASSO 2: INICIAR OUTRO LOOP QUE VAI RECEBER A VARIAVEL DO LOOP ANTERIOR E FAZER UMA CONTAGEM DE X NUMEROS
# PASSO 3: NO SEGUNDO LOOP, ELE IRÁ ADICIONAR NA LISTA O NUMERO DE ACORDO COM O SEU VALOR. EX.: 3 = 3, 3, 3
# PASSO 4: FAZER UM CONTADOR PARA AMARZENAR A QUANTIDADE DE NUMEROS
# PASSO 5: IMPRIMIR A MENSAGEM FORMATADA E LOGO ABAIXO A SEQUENCIA DENTRO DA LISTA


# SUBPROGRAMA
# PROGRAMA PRINCIPAL
caso = 1
while True:
    try:
        sequencia = []
        contarSequencia = 0
        tamanhoDaSequencia = int(input())

        while True:
            if contarSequencia >= tamanhoDaSequencia and tamanhoDaSequencia != 0:
                break
            while contarSequencia < tamanhoDaSequencia:
                sequencia.append(tamanhoDaSequencia)
                contarSequencia += 1
            tamanhoDaSequencia -= 1
            contarSequencia = 0

        sequencia.append(0)
        sequenciaOrdenada = sorted(sequencia)
        sequenciaString = (str(numero) for numero in sequenciaOrdenada)
        sequenciaString = ' '.join(sequenciaString)

    except EOFError:
        break
    else:
        if len(sequenciaOrdenada) <= 2:
            print('Caso %d: %d numero' % (caso, len(sequenciaOrdenada)))
        else:
            print('Caso %d: %d numeros' % (caso, len(sequenciaOrdenada)))
        print(sequenciaString)
        caso += 1
        print()

File: C/BEGINNER/2028 Sequência de Sequência.c
#include <stdio.h>

int main()
{
    int numSequencia, cont, caso, i, j;
    caso = 1;
    while (scanf("%d", &numSequencia) != EOF)
    {
        if (numSequencia == 0)
        {
            printf("Caso %d: 1 numero\n", caso);
            printf("0\n");
        }
        else if (numSequencia == 1)
        {
            printf("Caso %d: 2 numeros\n", caso);
            printf("0 1\n");
        }
        else
        {
            cont = 1;
            for (i=1; i <= numSequencia; i++)
            {
                for (j=i; j != 0; j--)
                {
                    cont += 1;
                }
            }
            printf("Caso %d: %d numeros\n", caso, cont);
            printf("0 ");
            for (i=1; i <= numSequencia; i++)
            {
                for (j=i; j != 0; j--)
                {
                    if ((j == 1) && (i == numSequencia))
                        break;
                    printf("%d ", i);
                }
            }
            printf("%d\n", numSequencia);
        }
        printf("\n");
        caso += 1;
    }
    return 0;
}
